scripts/scr_enemy_ai_a/scr_enemy_ai_a.gml

```diff
@@ -3,10 +3,14 @@ function scr_enemy_ai_a() {
 
 	// guardsmen hop from planet to planet
 	if (p_guardsmen[1]+p_guardsmen[2]+p_guardsmen[3]+p_guardsmen[4]>0) and (present_fleet[2]>0){
-	    var o,mx,cr,tr;o=0;mx=0;cr=0;tr=0;
-	    repeat(4){o+=1;if (p_guardsmen[o]>0) and (cr=0) then cr=o;}
+	    var o=0,mx=0,cr=0,tr=0;
+	    repeat(planets){
+	    	o+=1;
+	    	if (p_guardsmen[o]>0) and (cr=0) then cr=o;
+	    }
 	    o=0;
-	    repeat(4){o+=1;
+	    repeat(planets){
+	    	o+=1;
 	        if (p_orks[o]+p_tau[o]+p_chaos[o]+p_traitors[o]+p_tyranids[o]+p_necrons[o]>mx){
```

Comment:
While here, might make sense to include `p_eldar` in the checks or even wrap the logic into a function that gives you the average threat level of non-imperium aligned forces to test against.